sdk/identity/azure-identity/test/ut/chained_token_credential_test.cpp

```diff
@@ -110,7 +114,9 @@ TEST(ChainedTokenCredential, Logging)
     ChainedTokenCredential cred({});
     EXPECT_EQ(log.size(), LogMsgVec::size_type(1));
     EXPECT_EQ(log[0].first, Logger::Level::Warning);
-    EXPECT_EQ(log[0].second, "Identity: ChainedTokenCredential: Created with 0 credentials.");
+    EXPECT_EQ(
+        log[0].second,
+        "Identity: ChainedTokenCredential: Created with EMPTY chain of credentials.");
```

Comment:
Is this a valid scenario? Why do we support this?

Reply:
I understand that. I mean why would they do that? Why do we allow that case?

Reply:
Customer code could be structured in a way that vector is formed in one place, and the chained token credential gets created in another place. Initializing it with empty list gives no advantage, because the case is similar to creating it with one credential that is unable to get token. I.e. successful creation does not guarantee successful token retrieval.
If you look at the other SDKs, there is no unity - Java and JS do allow empty lists, while .NET, Go and Python do not.
We shipped this credential not throwing on empty lists. If we want to change that, that'd be a breaking change, and I don't think it is worth it.